cmd/snap/cmd_buy.go

```diff
@@ -95,21 +93,17 @@ func buySnap(opts *store.BuyOptions) error {
 	snap := snaps[0]
 
 	opts.SnapID = snap.ID
-
-	if opts.Channel == "" {
-		opts.Channel = snap.Channel
-	}
-
+	opts.Channel = snap.Channel
```

Comment:
Still awkward to see this here. The user is not buying from a channel.

Not a blocker now that it's just an implementation detail, but we should fix it.